subliminal/providers/gestdown.py

```
@@ -0,0 +1,298 @@
+
+    def get_matches(self, video):
+        # series name
+        matches = guess_matches(video, {
+            'title': self.series,
+            'season': self.season,
+            'episode': self.episode,
+            'episode_title': self.title,
+            'release_group': self.version,
+        })
+
+        # resolution
+        if self.version and video.resolution and video.resolution in self.version.lower():
+            matches.add('resolution')
+        # other properties
+        if self.version:
+            matches |= guess_matches(
+                video,
+                guessit(self.version, {'type': 'episode'}),
+                partial=True,
+            )
+
+        return matches
+
+
+class GestdownProvider(Provider):
+    """Gestdown Provider."""
+    languages = {Language(lang) for lang in [
+        'ara', 'aze', 'ben', 'bos', 'bul', 'cat', 'ces', 'dan', 'deu', 'ell', 'eng',
+        'eus', 'fas', 'fin', 'fra', 'glg', 'heb', 'hrv', 'hun', 'hye', 'ind', 'ita',
+        'jpn', 'kor', 'mkd', 'msa', 'nld', 'nor', 'pol', 'por', 'ron', 'rus', 'slk',
+        'slv', 'spa', 'sqi', 'srp', 'swe', 'tha', 'tur', 'ukr', 'vie', 'zho',
+    ]}
+    video_types = (Episode,)
+    server_url = 'https://api.gestdown.info'
+    subtitle_class = GestdownSubtitle
+
+    def __init__(self):
+        self.session = None
+
+    def initialize(self):
+        self.session = Session()
+        self.session.headers['User-Agent'] = self.user_agent
+        self.session.headers['accept'] = 'application/json'
+
+    def terminate(self):
+        if self.session is None:
+            return
+        self.session.close()
+
+    def _search_show_id(self, series, series_tvdb_id=None):
+        """Search the show id from the `series`.
+
+        :param str series: series of the episode.
+        :param int series_tvdb_id: tvdb id of the series.
+        :return: the show id, if found.
+        :rtype: int
+
+        """
+        if self.session is None:
+            return
```

maybe we could raise an exception when the provider isn't initialized but is attempted to be used?

---

That would be better, you are right!
Although that is a problem that is present in all the providers, so it would be better to create a specific `Exception` for it and correct all the provider. I think this is out of scope of this PR.

---

Ah yes, can you open an issue please?